junit-platform-commons/src/main/java/org/junit/platform/commons/support/scanning/ClasspathScanner.java

```diff
@@ -62,6 +64,55 @@ public interface ClasspathScanner {
 	 */
 	List<Class<?>> scanForClassesInClasspathRoot(URI root, ClassFilter classFilter);
 
+	/**
+	 * Find all {@linkplain org.junit.platform.commons.support.Resource resources} in the supplied classpath {@code root}
+	 * that match the specified {@code resourceFilter} predicate.
+	 *
+	 * <p>The classpath scanning algorithm searches recursively in subpackages
+	 * beginning with the root of the classpath.
+	 *
+	 * @param basePackageName the name of the base package in which to start
+	 * scanning; must not be {@code null} and must be valid in terms of Java
+	 * syntax
+	 * @param resourceFilter the resource type filter; never {@code null}
+	 * @return a list of all such resources found; never {@code null}
+	 * but potentially empty
+	 * @deprecated Please implement
+	 * {@link #scanForResourcesInPackage(String, ResourceFilter)} instead
+	 */
+	@API(status = DEPRECATED, since = "6.0")
+	@Deprecated(since = "6.0", forRemoval = true)
+	@SuppressWarnings({ "removal", "unused" })
+	default List<org.junit.platform.commons.support.Resource> scanForResourcesInPackage(String basePackageName,
+			Predicate<org.junit.platform.commons.support.Resource> resourceFilter) {
+		throw new UnsupportedOperationException("Implement scanForResourcesInPackage(String, ResourceFilter) instead");
+	}
+
```

Comment:
I've kept these methods for backwards compatibility with existing `ClasspathScanner` implementations. They will be removed in a separate PR: #4918.